bentoml/_internal/io_descriptors/base.py

```diff
@@ -5,83 +5,81 @@
 from abc import abstractmethod
 from typing import TYPE_CHECKING
 
-from starlette.requests import Request
-from starlette.responses import Response
-
 if TYPE_CHECKING:
     from types import UnionType
 
+    from typing_extensions import Self
+    from starlette.requests import Request
+    from starlette.responses import Response
+
     from ..types import LazyType
     from ..context import InferenceApiContext as Context
+    from ..service.openapi.specification import Schema
+    from ..service.openapi.specification import Response as OpenAPIResponse
+    from ..service.openapi.specification import Reference
+    from ..service.openapi.specification import RequestBody
 
+    InputType = (
+        UnionType
+        | t.Type[t.Any]
+        | LazyType[t.Any]
+        | dict[str, t.Type[t.Any] | UnionType | LazyType[t.Any]]
+    )
 
-IOPyObj = t.TypeVar("IOPyObj")
 
+IOType = t.TypeVar("IOType")
 
-_T = t.TypeVar("_T")
 
-
-class IODescriptor(ABC, t.Generic[IOPyObj]):
+class IODescriptor(ABC, t.Generic[IOType]):
     """
     IODescriptor describes the input/output data format of an InferenceAPI defined
     in a :code:`bentoml.Service`. This is an abstract base class for extending new HTTP
     endpoint IO descriptor types in BentoServer.
     """
 
     HTTP_METHODS = ["POST"]
+
     _init_str: str = ""
 
-    def __new__(cls: t.Type[_T], *args: t.Any, **kwargs: t.Any) -> _T:
+    _mime_type: str
+
+    def __new__(cls: t.Type[Self], *args: t.Any, **kwargs: t.Any) -> Self:
         self = super().__new__(cls)
-        arg_strs = tuple(repr(i) for i in args) + tuple(
-            f"{k}={repr(v)}" for k, v in kwargs.items()
-        )
-        setattr(self, "_init_str", f"{cls.__name__}({', '.join(arg_strs)})")
+        # default mime type is application/json
+        self._mime_type = "application/json"
+        self._init_str = cls.__qualname__
```

Comment:
reason behind this change is that the inferenceAPI name doesn't look nice on OpenAPI
![Screenshot 2022-08-02 at 20 25 26](https://user-images.githubusercontent.com/29749331/182517525-0fdae9c2-71c6-4729-9308-ad6297ecea0b.png)